src/FuncVersion.ts

```diff
@@ -4,7 +4,7 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { IActionContext, IAzureQuickPickItem, IAzureQuickPickOptions } from '@microsoft/vscode-azext-utils';
-import { localize } from './localize';
+import { localize, recommended } from './localize';
```

Comment:
Putting the localized strings in the `localize.ts` file isn't really the intended usage, but also, this contradicts with the way the new `l10n` API is going to be used. It will be easiest to keep all of the strings in their original locations in the source code, rather than clustered in a file full of constants.

Reply:
Hmm, so I only added the localize constants that were being used in multiple files here.  Previously localize constants that fit this bill were being shown in `constants.ts` but it was starting to look really cluttered to me so I figured placing the localize constants in a separate file might make more sense/be cleaner.

Out of curiosity, will the `l10n` API not be caching the localize string by ID (i.e. the first parameter of the localize call)?  Would it make sense for me to remove mine and all previous localize constants as well (previously part of the migration from `constants.ts`?  Thanks :)

Reply:
Yeah, I think they should all be removed from `constants.ts` as well. The `l10n` API uses the string itself as the key, plus any comment you add for applying uniqueness as needed or clarifying things for the localizers. Not needing an ID I think is a good improvement; there will doubtless be some iteration though on fixing bugs that arise from loss of uniqueness in the keys.

Reply:
Okay, that makes a lot of sense, thank you for the explanation 🙂

Reply:
> Yeah, I think they should all be removed from `constants.ts` as well. The `l10n` API uses the string itself as the key, plus any comment you add for applying uniqueness as needed or clarifying things for the localizers. Not needing an ID I think is a good improvement; there will doubtless be some iteration though on fixing bugs that arise from loss of uniqueness in the keys.

I'm not following why we would move all of the localized strings out of `constants.ts`. Even if localize is using the string as the key, does its location in source get stored as metadata as well? Because if not, it might be easier to change all of our localized strings in one file (especially if some of those strings are reused, such as a "Yes" label).

Definitely not in `localize.ts`, but maybe in some file called `constants-nls.ts` or something?

Reply:
The old `vscode-nls` way required a parallel file tree to your whole code base with a file containing translations for every file in source that had strings. Multiply that by language and you have a colossal number of files.

The new `l10n` way puts all those strings into one file (per language), using the original string from the source as a key (plus comment if applicable), with the value being whatever the localized string is.

With the new `l10n` API, putting the strings into `constants.ts` (or, `localize.ts`) is not going to make the actual process of localization any easier--but it is going to mean anytime I want to see what a string value is, I have to <kbd>F12</kbd> over to the constants file instead of being able to just read it inline.

Reply:
That being said, for strings we expect to change frequently, and/or that we want to all stay in sync, I think it does make sense to use a file with constants (though I wouldn't use `constants.ts`, and not `localize.ts` either since that's `vscode-nls`-specific.)